specification/loadtestservice/LoadTestService/models.tsp

```diff
@@ -0,0 +1,1049 @@
+  @doc("Type of test. e.g. Url or JMX")
+  testType?: TestType;
+}
+
+@doc("Configuration for quick load test")
+model OptionalLoadTestConfig {
+  @doc("Test URL. Provide the complete HTTP URL. For example, https://contoso-app.azurewebsites.net/login")
+  endpointUrl?: string;
+
+  @doc("Throughput required (requests per second).")
+  @added(APIVersions.v2023_04_01_preview)
+  rps?: int32;
+
+  @doc("Maximum response time in milliseconds of the API/endpoint.")
+  @added(APIVersions.v2023_04_01_preview)
+  maxResponseTime?: int32;
+
+  @doc("No of concurrent virtual users.")
+  virtualUsers?: int32;
+
+  @doc("Ramp up time in seconds.")
+  rampUpTime?: int32;
+
+  @doc("Test run duration in seconds.")
+  duration?: int32;
+}
+
+@doc("The input artifacts for the test.")
+model TestInputArtifacts {
+  @doc("File info")
+  configFileInfo?: TestFileInfo;
+
+  @doc("File info")
+  testScriptFileInfo?: TestFileInfo;
+
+  @doc("File info")
+  userPropFileInfo?: TestFileInfo;
+
+  @doc("File info")
+  inputArtifactsZipFileInfo?: TestFileInfo;
+
+  @added(APIVersions.v2023_04_01_preview)
+  @doc("The config json file for url based test")
+  urlTestsConfigFileInfo?: TestFileInfo;
+
+  @doc("Additional supported files for the test run")
+  @visibility("read")
+  additionalFileInfo?: TestFileInfo[];
+}
+
+@doc("File info")
+model FileInfo {
+  @doc("File URL.")
+  @visibility("read")
+  url?: string;
+
+  @doc("File type")
+  @visibility("read")
+  fileType?: FileType;
+
+  @doc("Expiry time of the file (ISO 8601 literal format)")
+  @visibility("read")
+  expireDateTime?: utcDateTime;
+
+  @doc("Validation status of the file")
+  @visibility("read")
+  validationStatus?: FileStatus;
+
+  @doc("Validation failure error details")
+  @visibility("read")
+  validationFailureDetails?: string;
+}
+
+@doc("Collection of tests")
+@friendlyName("TestsList")
+model TestsList is Azure.Core.Page<Test>;
+
+@doc("Test app components")
+model TestAppComponents {
+  @doc("""
+Azure resource collection { resource id (fully qualified resource Id e.g
+subscriptions/{subId}/resourceGroups/{rg}/providers/Microsoft.LoadTestService/loadtests/{resName})
+: resource object } 
+""")
+  components: Record<AppComponent>;
+
+  @doc("Test identifier")
+  @visibility("read")
+  testId?: string;
+
+  ...SystemMetaData;
+}
+
+@doc("""
+An Azure resource object (Refer azure generic resource model :
+https://docs.microsoft.com/en-us/rest/api/resources/resources/get-by-id#genericresource)
+""")
+model AppComponent {
+  @doc("""
+fully qualified resource Id e.g
+subscriptions/{subId}/resourceGroups/{rg}/providers/Microsoft.LoadTestService/loadtests/{resName}
+""")
+  @visibility("read")
+  resourceId: string;
+
+  @doc("Azure resource name, required while creating the app component.")
+  resourceName: string;
+
+  @doc("Azure resource type, required while creating the app component.")
+  resourceType: string;
+
+  @doc("Azure resource display name")
+  displayName: string;
```

Comment:
Is `displayName` required? I see that the descriptions for `resourceName` and `resourceType` say they are required, but no for `displayName`.

Reply:
Display name is not required for this model. Removing it from required list.